src/coreclr/System.Private.CoreLib/src/ILLink/ILLink.Descriptors.xml

```diff
@@ -25,6 +25,18 @@
     <type fullname="System.SZArrayHelper">
       <method name=".ctor" />
     </type>
+
+    <!-- Native hosting accesses managed methods in the ComponentActivator class. -->
+    <type fullname="Internal.Runtime.InteropServices.ComponentActivator" feature="Internal.Runtime.InteropServices.ComponentActivator.IsSupported" featurevalue="true" featuredefault="true">
+      <method name="LoadAssemblyAndGetFunctionPointer" />
+      <method name="GetFunctionPointer" />
+    </type>
+
+    <!-- Enables the .NET IJW host to load an in-memory module as a .NET assembly. -->
+    <type fullname="Internal.Runtime.InteropServices.InMemoryAssemblyLoader" feature="Internal.Runtime.InteropServices.InMemoryAssemblyLoader.IsSupported" featurevalue="true" featuredefault="true">
+      <method name="LoadInMemoryAssembly" />
+    </type>
```

Comment:
> feature="Internal.Runtime.InteropServices.InMemoryAssemblyLoader.IsSupported" featurevalue="true"

Does this mean that the entry points will be entirely trimmed away if `Internal.Runtime.InteropServices.ComponentActivator.IsSupported` / `Internal.Runtime.InteropServices.InMemoryAssemblyLoader.IsSupported` is set to `false`? If so, what error gets propagated to an attempt to consume them? Missing method, type load?

Reply:
hmmm. missed that in the last change. @vitek-karas?

Reply:
I actually don't know - we should try that (It probably depends, but hopefully the entire type goes away). It is definitely worth trying to see what error we get back into the native side - and if we can turn it into the new error code. Hopefully it's not just `E_FAIL`. If the experience is bad, maybe we should actually keep the entry points around just so that they provide better error experience.

Please also validate that this allows the trimmer to remove the code (in both cases)

Reply:
Reverted descriptor setting to an earlier checking that will do the right thing. Setting property to false will have the desired behavior where the native call will get to see the error codes specific to trimming (no warnings), True will generate warnings and will leave the method untouched

Reply:
If we think that rooting the entry points always is the right solution, then there should be no `featurevalue` or `featuredefault` attributes on these elements (in the descriptor).

I would also add a comment why we're rooting them always and that the body should be removed due to feature switches.